libraries/AP_InternalError/AP_InternalError.h

```diff
@@ -111,5 +111,16 @@ extern "C" {
     void AP_memory_guard_error(uint32_t size);
 }
 
+#if (defined(AP_WITH_SEMAPHORE_CONSISTENT_BUILDS_ENABLED) && AP_WITH_SEMAPHORE_CONSISTENT_BUILDS_ENABLED)
+
+// use dummy values for line for consistent builds even if code has
+// been moved around a bit:
+#define INTERNAL_ERROR(error_number) \
+    AP::internalerror().error(error_number, 387);
+
+#else
+
 #define INTERNAL_ERROR(error_number) \
     AP::internalerror().error(error_number, __LINE__);
```

Comment:
using __AP__LINE__ may be better? then we don't need ifdefs in lots of places

Reply:
Changed to use this method.  *Much* nicer delta.

I've tested that we still get correct line numbers when you don't use this option and get 17 when you do.